app/services/redmine_request_csv.rb

```diff
@@ -1,42 +1,41 @@
 class RedmineRequestCsv
-  def initialize
-    @url = create_url
-  end
+  INVESTMENT_PARAMS = [['f[]', 'project_id'], ['op[project_id]', '=']]
+                      .push(*RedmineInvestmentProjects.new.ids.map { |id| ['v[project_id][]', id] }).freeze
 
-  def request_redmine_for_entries
-    http_request
+  def initialize(investment_params = [])
+    @url = create_url(investment_params)
   end
 
-  private
-
-  def create_url
-    @url = URI('https://redmine.renuo.ch/time_entries/report.csv?' + query + key)
+  def execute_request
+    http_request(@url)
   end
 
-  def query
-    URI.encode_www_form([['utf8', '✓'], ['criteria[]', 'user'],
-                         ['f[]', 'spent_on'], ['op[spent_on]', '>='], ['v[spent_on][]', '2017-01-30'],
-                         ['f[]', ''],
-                         ['c[]', 'project'], ['c[]', 'spent_on'], ['c[]', 'user'],
-                         ['c[]', 'activity'], ['c[]', 'issue'], ['c[]', 'comments'],
-                         ['c[]', 'hours'],
-                         %w(columns month),
-                         ['criteria[]', '']].push(*addition_params))
-  end
+  private
 
-  def addition_params
-    []
+  def create_url(params)
+    URI('https://redmine.renuo.ch/time_entries/report.csv?' + query(params) + key)
   end
 
   def key
     '&key=' + ENV['REDMINE_API_KEY']
   end
 
-  def http_request
-    req = Net::HTTP::Get.new(@url)
+  def http_request(url)
+    req = Net::HTTP::Get.new(url)
 
-    Net::HTTP.start(@url.hostname, @url.port, use_ssl: true) do |http|
+    Net::HTTP.start(url.hostname, url.port, use_ssl: true) do |http|
       http.request(req)
     end
   end
```

Comment:
Aren't there [shortcuts](https://docs.ruby-lang.org/en/2.0.0/Net/HTTP.html#class-Net::HTTP-label-GET+by+URI) for this block notation?